src/Frontend/Validation.hs

```diff
@@ -7,26 +8,43 @@ module Frontend.Validation
     , validation
     , (<??>)
     , FieldParser
+    , satisfies
     )
 where
 
---import Data.String.Conversions (ST, cs)
 import Text.Digestive as TD
 import Text.Parsec as TP
+import Text.Parsec.Error
 
 import Frontend.Prelude
 
 type FieldParser a = Parsec String () a
 
 -- FIXME: Use (Error -> Html) instead of toHtml
-fieldValidation :: FieldParser a -> String -> Result (Html ()) a
-fieldValidation field t =
-    either (TD.Error . toHtml . show) TD.Success $ parse field "" (cs t)
+fieldValidation :: String -> FieldParser a -> String -> Result (Html ()) a
+fieldValidation name parser t =
```

Comment:
I'm afraid I started that, but now I am beginning to feel that we are using too many one-letter names...